Adds 3 scalars for use with NekRS simulations

ahuxford opened the pull request:
Mainly for use with NekRS RANS k-tau simulations where k and tau are scalar01 and scalar02. I added an additional scalar03 in case an additional passive scalar transport is of interest.

moosebuild commented:
Job [Test](https://civet.inl.gov/job/1007351/) on 8871f32 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/cardinal/docs/PRs/358/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1007354/) on 8871f32 wanted to post the following:


## Coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>ad3a8e</th>
      <th colspan=4>#358 8871f3</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>88.97%</td>
      <td>88.73%</td>
      <td>-0.23%</td>
      <td>12.50%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>5039</td>
      <td>5039</td>
      <td>-</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>625</td>
      <td>640</td>
      <td>+15</td>
      <td>14</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/cardinal/docs/PRs/358/coverage_diff/diff.html)

[Full coverage report](https://mooseframework.inl.gov/cardinal/docs/PRs/358/coverage)

## Warnings

- New new line coverage rate 12.50% is less than the suggested 90.0%



This comment will be updated on new commits.

aprilnovak commented:
This looks good! There's a few features we should add to be sure the documentation is up to date and one sanity check:

- [ ] An error test that throws the correct error if the user puts `scalar03`, but doesn't have 4 scalars - otherwise, this option would seg fault
- [ ] In the input parameters for some of the classes, we should add the new options to the documentation string, i.e. 

```
params.addRequiredParam<MooseEnum>("field",
                                     getNekFieldEnum(),
                                     "Field to integrate; "
                                     "options: velocity_x, velocity_y, velocity_z, "
                                     "velocity_component, velocity, temperature, pressure, unity");
```

- [ ] Add the new options to `field_specs.md` so that they show up in the documentation (also add the caveat that we're not dimensionalizing these fields, in `nondimensional.md`

ahuxford commented:
Yes good points. I'll update this PR with the changes so we add everything at once for these added scalars.

ahuxford commented:
I think I addressed all of your feedback.

For the tests, I used `test/tests/nek_errors/invalid_field` as a template to test Nek postProcessors and Nek UserObjects.

I did not add the scalars to the `NekRSStandaloneProblem` options for `outputs` yet, so there's no test for that (unlike the test for temperature). 

I do plan to add the scalars to the `outputs` options in a future PR though, because that will be very useful.

moosebuild commented:
Job [Test nekRS](https://civet.inl.gov/job/1009652/) on 888be89 : invalidated by @aprilnovak

random nek failure

ahuxford commented on src/postprocessors/NekPostprocessor.C:
```diff
@@ -54,6 +54,18 @@ NekPostprocessor::checkValidField(const field::NekFieldEnum & field) const
   if (!nekrs::hasTemperatureVariable() && field == field::temperature)
     mooseError("This postprocessor cannot set 'field = temperature' "
                "because your Nek case files do not have a temperature variable!");
+
```
Yes that'd be better. I'll move the errors to solutionPointer. I'll move the temperature over to there also right now.

ahuxford commented:
I think this should be ready once the tests finish.

moosebuild commented:
Job [Test nekRS](https://civet.inl.gov/job/1010494/) on 20d23cf : invalidated by @aprilnovak

random nek failure?

aprilnovak commented:
Thanks for this contribution!
